src/dune/build_system.ml

```diff
@@ -468,6 +510,15 @@ let compute_targets_digest targets =
   | l -> Some (Digest.generic l)
   | exception (Unix.Unix_error _ | Sys_error _) -> None
 
+let find_rule_source_dir ({ Rule.action; _ } as rule) =
+  let _, src_dir =
+    match Path.Build.Set.find ~f:(fun _ -> true) action.targets with
```

Comment:
How is this dir different from `rule.dir`?

Reply:
`rule.dir` is the path to the built target from the source directory, eg `_build/default/src/dune`, while `find_rule_source_dir` is the "corresponding" source directory, eg in that case `src/dune`. This does not just trim the `_build/default` prefix but also handle cases such as `_build/default/src/dune/.dune.objs`, with build specific subdirectories removed to become `src/dune` too.

These source directories help locate which VCS repository targets must be associated too, so we can index produced files by commit.

Reply:
Just saw your response, sorry for the delay. Actually, I just meant the `match Path.Build.Set.find ..` bit. I think that it would be simpler to implement the desired `find_rule_source_dir` using `rule.dir`.

Reply:
Ah you're right, sorry, thanks !